src/tools/scripts/set_lighthouse_urls.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+LIGHTHOUSE_CONFIG_FILE="../.github/lighthouse/lighthouse-config-dev.json"
+LIGHTHOUSE_PROD_CONFIG_FILE="../.github/lighthouse/lighthouse-config-prod.json"
+
+# Usage info
+show_help() {
+cat << EOH
+Usage: ${0##*/} [-p]
+Get a list of URLs to run a lighthouse test on.
+If a commit is given then only URLs changes are tested otherwise all.
+
+    -h   display this help and exit
+    -p   get all production URLs from sitemap.xml
+EOH
+}
+
+OPTIND=1 # Reseting is good practive
+production=0
+while getopts "h?p" opt; do
+    case "$opt" in
+    h|\?)
+        show_help
+        exit 0
+        ;;
+    p)  production=1
+        ;;
+    esac
+done
+shift "$((OPTIND-1))" # Discard the options and sentinel --
+
+# This script must be run from src directory
+if [ -d "src" ]; then
+  cd src || exit
+fi
+
+LIGHTHOUSE_URLS=""
+
+# Set some URLs that should always be checked on pull requests
+# to ensure basic coverage
+BASE_URLS=$(cat <<-END
+http://127.0.0.1:8080/en/2019/
+http://127.0.0.1:8080/en/2019/javascript
+http://127.0.0.1:8080/en/2020/
+END
+)
+
+if [ "${production}" == "1" ]; then
+    
+    # Get the production URLs from the production sitemap
+    LIGHTHOUSE_URLS=$(curl -s https://almanac.httparchive.org/sitemap.xml | grep "<loc" | grep -v static | sed 's/ *<loc>//g' | sed 's/<\/loc>//g')
+
+    # Switch to the Production Config file
+    LIGHTHOUSE_CONFIG_FILE="${LIGHTHOUSE_PROD_CONFIG_FILE}"
+
+elif [ "${RUN_TYPE}" == "pull_request" ] && [ "${COMMIT_SHA}" != "" ]; then
+
+    # If this is part of pull request then get list of files as those changed
+    # Uses similar logic to GitHub Super Linter
+    # First checkout main to get list of differences
+    git pull --quiet
+    git checkout main
+    # Then get the changes
+    CHANGED_FILES=$(git diff --name-only "main...${COMMIT_SHA}" --diff-filter=d content templates | grep -v base.html | grep -v ejs | grep -v base_ | grep -v toc.html | grep -v sitemap)
+    # Then back to the pull request changes
+    git checkout --progress --force "${COMMIT_SHA}"
+
+    # Transform the files to http://127.0.0.1:8080 URLs
+    LIGHTHOUSE_URLS=$(echo "${CHANGED_FILES}" | sed 's/src\/content/http:\/\/127.0.0.1:8080/g' | sed 's/\.md//g' | sed 's/\/base\//\/en\//g' | sed 's/src\/templates/http:\/\/127.0.0.1:8080/g' | sed 's/\.html//g' | sed 's/_/-/g' | sed 's/\/2019\/accessibility-statement/\/accessibility-statement/g' )
+    if [ "${LIGHTHOUSE_URLS}" = "" ]; then
+        LIGHTHOUSE_URLS="${BASE_URLS}"
+    else
+        LIGHTHOUSE_URLS=$( echo -e "${LIGHTHOUSE_URLS}\n${BASE_URLS}")
+    fi
+
+else
+
+    # Else test every URL (except PDFs) in the sitemap
+    LIGHTHOUSE_URLS=$(grep loc templates/sitemap.xml | grep -v static | sed 's/ *<loc>//g' | sed 's/<\/loc>//g' | sed 's/https:\/\/almanac.httparchive.org/http:\/\/127.0.0.1:8080/g')
```

Comment:
Many sed instances can be removed in favor or a single grep as noted above.

Reply:
Same answer as above. Will keep with simplicity over single commands.